pkg/database/upsertCluster.go

```diff
@@ -3,12 +3,31 @@ package database
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"reflect"
 
 	"github.com/stolostron/search-indexer/pkg/model"
 	"k8s.io/klog/v2"
 )
 
+func (dao *DAO) DeleteCluster(ctx context.Context, clusterName string) {
+	clusterUID := string("cluster__" + clusterName)
+	// Delete resources for cluster from resources table from DB
+	_, err := dao.pool.Exec(ctx, "DELETE FROM search.resources WHERE cluster=$1", clusterName)
+	checkError(err, fmt.Sprintf("Error deleting resources from search.resources for clusterName %s.", clusterName))
+
+	// Delete edges for cluster from DB
+	_, err = dao.pool.Exec(ctx, "DELETE FROM search.edges WHERE cluster=$1", clusterName)
+	checkError(err, fmt.Sprintf("Error deleting resources from search.edges for clusterName %s.", clusterName))
+
+	// Delete cluster node from DB
+	_, err = dao.pool.Exec(ctx, "DELETE FROM search.resources WHERE uid=$1", clusterUID)
+	checkError(err, fmt.Sprintf("Error deleting cluster %s from search.resources.", clusterName))
```

Comment:
- Do we need to make these 3 operations in a single transaction? Would partial deletes cause a problem? For example if we delete the nodes but can't delete the edges?
- We should retry if there's any errors. This is the only chance we get to delete resources cleanly. In the future, we'll need to add a validation function to cleanup if we miss this event.

Reply:
Made into a single transaction and added retry. How often should the retry be?

Reply:
Tests needs to be updated

Reply:
This retry doesn't need to be as aggressive as the DB connection, but adding another env is unnecessary complexity.

What if we increase the default MAX_BACKOFF_MS to 300000 (5 min)
Then for DB connection we use `MAX_BACKOFF_MS/10` (30 sec)

Would that be confusing? I don't expect end users will change this, so maybe all we need is to add a comment in the code for us to remember later.

Reply:
Done